pkg/kotsadmsnapshot/backup.go

```diff
@@ -191,254 +201,512 @@ func CreateApplicationBackup(ctx context.Context, a *apptypes.App, isScheduled b
+		return "", errors.Wrap(err, "failed to create clientset")
 	}
 
-	isKurl, err := kurl.IsKurl(clientset)
+	ctrlClient, err := k8sutil.GetKubeClient(ctx)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to check if cluster is kurl")
+		return "", fmt.Errorf("failed to get kubeclient: %w", err)
 	}
 
-	kotsadmNamespace := util.PodNamespace
-	appsSequences := map[string]int64{}
-	appVersions := map[string]string{}
-	includedNamespaces := []string{kotsadmNamespace}
-	excludedNamespaces := []string{}
-	backupAnnotations := map[string]string{}
-	backupOrderedResources := map[string]string{}
-	backupHooks := velerov1.BackupHooks{
-		Resources: []velerov1.BackupResourceHookSpec{},
-	}
-	// non-supported fields that are intentionally left out cuz they might break full snapshots:
-	// - includedResources
-	// - excludedResources
-	// - labelSelector
+	veleroClient, err := veleroclientv1.NewForConfig(cfg)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to create velero clientset")
+	}
 
-	appNamespace := kotsadmNamespace
-	if os.Getenv("KOTSADM_TARGET_NAMESPACE") != "" {
-		appNamespace = os.Getenv("KOTSADM_TARGET_NAMESPACE")
+	metadata, err := getInstanceBackupMetadata(ctx, k8sClient, ctrlClient, veleroClient, cluster, isScheduled)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get instance backup metadata")
 	}
-	if appNamespace != kotsadmNamespace {
-		includedNamespaces = append(includedNamespaces, appNamespace)
+
+	appVeleroBackup, err := getAppInstanceBackupSpec(k8sClient, metadata)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get app instance backup spec")
 	}
 
-	if isKurl {
-		includedNamespaces = append(includedNamespaces, "kurl")
+	veleroBackup, err := getInstanceBackupSpec(ctx, k8sClient, metadata, appVeleroBackup != nil)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get instance backup specs")
+	}
+
+	err = excludeShutdownPodsFromBackup(ctx, k8sClient, veleroBackup)
+	if err != nil {
+		logger.Errorf("Failed to exclude shutdown pods from backup: %v", err)
+	}
+
+	if appVeleroBackup != nil {
+		err = excludeShutdownPodsFromBackup(ctx, k8sClient, appVeleroBackup)
+		if err != nil {
+			logger.Errorf("Failed to exclude shutdown pods from application backup: %v", err)
+		}
+	}
+
+	logger.Infof("Creating instance backup CR %s", veleroBackup.Name)
+	backup, err := veleroClient.Backups(metadata.backupStorageLocationNamespace).Create(ctx, veleroBackup, metav1.CreateOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, "failed to create velero backup")
+	}
+
+	if appVeleroBackup != nil {
+		logger.Infof("Creating instance app backup CR %s", appVeleroBackup.Name)
+		_, err := veleroClient.Backups(metadata.backupStorageLocationNamespace).Create(ctx, appVeleroBackup, metav1.CreateOptions{})
+		if err != nil {
+			return "", errors.Wrap(err, "failed to create application velero backup")
+		}
+	}
+
+	return backup.Name, nil // TODO(improveddr): return metadata.BackupName
```

Comment:
We're only returning the `Infrastructure Backup` name for now?

Reply:
ya. we need to implement the api before we can return the backup name from the annotation